lib/upipe-blackmagic/upipe_blackmagic_source.cpp

```diff
@@ -821,8 +821,10 @@ static int upipe_bmd_src_set_uri(struct upipe *upipe, const char *uri)
     }
 
     const char *model_name;
-    if (deckLink->GetModelName(&model_name) == S_OK)
+    if (deckLink->GetModelName(&model_name) == S_OK) {
         upipe_notice_va(upipe, "detected card type %s", model_name);
+        free((void*)model_name);
```

Comment:
It's declared as const char ** in the header:
virtual HRESULT GetModelName (/* out */ const char **modelName) = 0;

Here is some code copy/pasted from blackmagic own example Capture:

        deckLink->GetModelName((const char**)&m_deckLinkName);
        free(m_deckLinkName);
        char* m_deckLinkName;

As you see they cast the other way around...